src/main/java/com/google/sps/servlets/Util.java

```diff
@@ -0,0 +1,148 @@
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import org.apache.commons.io.FileUtils;
+
+class Util {
+
+  private static final Gson GSON_OBJECT = new Gson();
+  private static final String CLIENT_SECRET_FILE = "/client_info.json";
+  private static final String API_KEY_FILE = "/api_key.txt";
+  private static final OkHttpClient client = new OkHttpClient();
+  private static final String INVALID_ACCESS_TOKEN = "INVALID";
+  private static final String EMPTY_REFRESH_TOKEN = "";
+  private static final String EMPTY_API_KEY = "";
+  private static final String EMPTY_PAGE_TOKEN = "";
+  private static final String ALL_DEVICES_ENDPOINT = "https://www.googleapis.com/admin/directory/v1/customer/my_customer/devices/chromeos";
+  private static final String DEFAULT_MAX_DEVICES = "200"; //is limited to effectively 200
+  private static final String DEFAULT_SORT_ORDER = "ASCENDING";
+  private static final String DEFAULT_PROJECTION = "FULL";
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final String API_KEY = getAPIKey(); 
+
+  public static List<ChromeOSDevice> getAllDevices(String userId) throws IOException {
+    final String refreshToken = getRefreshToken(userId);
+    File file = new File(Util.class.getResource(CLIENT_SECRET_FILE).getFile());
+    final GoogleClientSecrets clientSecrets =
+    GoogleClientSecrets.load(
+      JacksonFactory.getDefaultInstance(), new FileReader(file));
+    final String clientId = clientSecrets.getDetails().getClientId();
+    final String clientSecret = clientSecrets.getDetails().getClientSecret();
+    final String accessToken = getAccessToken(refreshToken, clientId, clientSecret);
+    ListDeviceResponse resp = getDevicesResponse(EMPTY_PAGE_TOKEN, accessToken);
+    final List<ChromeOSDevice> allDevices = new ArrayList<>(resp.getDevices());
+    while (resp.hasNextPageToken()) {
+      final String pageToken = (String) resp.getNextPageToken();
+      resp = getDevicesResponse(pageToken, accessToken);
+      allDevices.addAll(resp.getDevices());
+    }
+    System.out.println("finished");
+    System.out.println(getAPIKey());
+    return allDevices;
+  }
+
+  public static String getAPIKey() {
+    try {
+      File file = new File(Util.class.getResource(API_KEY_FILE).getFile());
+      String str = FileUtils.readFileToString(file);
+      return str;
+    } catch (IOException e) {
+      System.out.println(e.getMessage());
+    }
+  return EMPTY_API_KEY;
+  }
+
+  private static String getRefreshToken(String userId) throws IOException {
+    Query query = new Query("RefreshToken").setFilter(FilterOperator.EQUAL.of("userId", userId));
+    PreparedQuery results = datastore.prepare(query);
+    String refreshToken = EMPTY_REFRESH_TOKEN;
+    for (final Entity entity : results.asIterable()) {
+      refreshToken = (String) entity.getProperty("refreshToken");
+      break;
+    }
```

Comment:
If you just want to get the first result back I think you can use [results.asSingleEntity()](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery#method-summary) which will give you the first entity without having to use a for-loop and break immediately.